### PR TITLE
Implemented SearchParameters::escapeFilterValue()

### PR DESCRIPTION
**Pull request by Toflar**

### CODE REVIEW

File: src/SearchParameters.php [hunk 1]
```diff
@@ -48,6 +48,15 @@ public static function create(): self
         return new self();
     }
 
+    public static function escapeFilterValue(string|int|float|bool $value): string
+    {
+        return match (true) {
+            \is_bool($value) => $value ? 'true' : 'false',
```
Comment (alexander-schranz) [inline src/SearchParameters.php:54]: This need to be `1` or `0`:

```suggestion
            \is_bool($value) => $value ? '1' : '0',
```

```
1) Schranz\Search\SEAL\Adapter\Loupe\Tests\LoupeSearcherTest::testEqualConditionWithBoolean
Loupe\Loupe\Exception\FilterFormatException: Col 12: Error: Expected valid string or float value, got 'true'

/Users/alexanderschranz/Documents/Projects/schranz-search/packages/seal-loupe-adapter/vendor/loupe/loupe/src/Internal/Filter/Parser.php:307
/Users/alexanderschranz/Documents/Projects/schranz-search/packages/seal-loupe-adapter/vendor/loupe/loupe/src/Internal/Filter/Parser.php:156
/Users/alexanderschranz/Documents/Projects/schranz-search/packages/seal-loupe-adapter/vendor/loupe/loupe/src/Internal/Filter/Parser.php:145
/Users/alexanderschranz/Documents/Projects/schranz-search/packages/seal-loupe-adapter/vendor/loupe/loupe/src/Internal/Filter/Parser.php:195
/Users/alexanderschranz/Documents/Projects/schranz-search/packages/seal-loupe-adapter/vendor/loupe/loupe/src/Internal/Filter/Parser.php:65
/Users/alexanderschranz/Documents/Projects/schranz-search/packages/seal-loupe-adapter/vendor/loupe/loupe/src/Internal/Search/Searcher.php:395
/Users/alexanderschranz/Documents/Projects/schranz-search/packages/seal-loupe-adapter/vendor/loupe/loupe/src/Internal/Search/Searcher.php:65
/Users/alexanderschranz/Documents/Projects/schranz-search/packages/seal-loupe-adapter/vendor/loupe/loupe/src/Internal/Engine.php:157
/Users/alexanderschranz/Documents/Projects/schranz-search/packages/seal-loupe-adapter/vendor/loupe/loupe/src/Loupe.php:74
/Users/alexanderschranz/Documents/Projects/schranz-search/packages/seal-loupe-adapter/src/LoupeSearcher.php:117
/Users/alexanderschranz/Documents/Projects/schranz-search/packages/seal/src/Search/SearchBuilder.php:98
/Users/alexanderschranz/Documents/Projects/schranz-search/packages/seal/src/Testing/AbstractSearcherTestCase.php:310
```

Comment (alexander-schranz) [reply src/SearchParameters.php:54]: @Toflar all other cases seems to work fine now, atleast I did not yet stumble over other special chars issues.

File: tests/Functional/SearchTest.php [hunk 2]
```diff
@@ -681,6 +681,39 @@ public function testEqualFilter(string $filter, array $expectedHits): void
         ]);
     }
 
+    public function testEscapingFilterValues(): void
+    {
+        $configuration = Configuration::create()
+            ->withFilterableAttributes(['title'])
+        ;
+        $title = "The 17\" O'Conner && O`Series \n OR a || 1%2 book?";
```
Comment (alexander-schranz) [inline tests/Functional/SearchTest.php:689]: In redisearch i did stumble over some more issues so I extended this string with:

```suggestion
        $title = "^The 17\" O'Conner && O`Series \n OR a || 1%2 \n book? \r \twhat \\ text // ok? end$";
```

**Comment by Toflar** [comment]
Boolean handling hasn't been implemented at all 😅 Should be better now.

**Comment by alexander-schranz** [comment]
I see I currently did on 0.4 convert bool to 1 or 0 which did for my test cases work. But true or false as keywords sounds good.

**Comment by Toflar** [comment]
It works because SQLite does that internally anyway and it will continue to work. But I think it's still better to document and use explicit `true` and `false` values.

File: tests/Functional/SearchTest.php [hunk 3]
```diff
@@ -681,6 +681,47 @@ public function testEqualFilter(string $filter, array $expectedHits): void
...
 
+    public function testEscapeFilterValues(): void
+    {
+        $configuration = Configuration::create()
+            ->withFilterableAttributes(['title', 'published'])
+        ;
+
+        $document = [
+            'id' => 42,
+            'title' => "^The 17\" O'Conner && O`Series \n OR a || 1%2 \r\n book? \r \twhat \\ text // ok? end$",
```
Comment (alexander-schranz) [inline tests/Functional/SearchTest.php:692]:
```suggestion
            'title' => "^The 17\" O'Conner && O`Series \n OR a || 1%2 1~2 1*2 \r\n book? \r \twhat \\ text: }{ )( ][ - + // \n\r ok? end$",
```

**Pull request merged**